### ISSUE TITLE
golangci-lint docker image does not work with go 1.22.1

### ISSUE THREAD

**Issue by fjsnogueira** [golangci/golangci-lint#4490]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've read the typecheck section of the FAQ (https://golangci-lint.run/usage/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

golangci-lint docker image does not work with go 1.22.1

![image](https://github.com/golangci/golangci-lint/assets/1634910/d3e778aa-a756-46c9-bb6c-dc7bad292597)

### Version of golangci-lint

<details>
golangci/golangci-lint:v1.56.2-alpine
</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

the docker image of golangci/golangci-lint:v1.56.2-alpine ships with go 1.22 (not 1.22.1) that's why it fails.

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

**Comment by ldez** [comment]
Hello,

It's a duplicate of #4239
